### PR TITLE
Refactor signing and integrate entangled/mobile

### PR DESCRIPTION
**Pull request by chrisdukakis**
# Description

- Refactor signing package
- Use entangled for signing in react native

## Type of change

Please delete options that are not relevant.

- [x] Enhancement (non-breaking change which adds functionality)

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/signing/android/src/main/java/org/iota/mobile/Interface.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.iota.mobile;
```
Comment (rajivshah3) [inline packages/signing/android/src/main/java/org/iota/mobile/Interface.java]: Should this be `org.iota.Signing`?

File: packages/signing/android/src/main/java/org/iota/mobile/Converter.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package org.iota.mobile;
```
Comment (rajivshah3) [inline packages/signing/android/src/main/java/org/iota/mobile/Converter.java]: Should this be `org.iota.Signing`?

File: packages/signing/android/src/main/java/org/iota/mobile/Interface.java (hunk 1 above)
Comment (chrisdukakis) [reply packages/signing/android/src/main/java/org/iota/mobile/Interface.java]: It should and we'll have to separate pow, hashing and signing over the next days.

Comment (chrisdukakis) [reply packages/signing/android/src/main/java/org/iota/mobile/Interface.java]: renamed in https://github.com/iotaledger/iota.js/pull/355/commits/a09e7908381b1ed8dab7d0b7f7c1327f23455f89

File: packages/signing/android/src/main/java/org/iota/mobile/Converter.java (hunk 2 above)
Comment (chrisdukakis) [reply packages/signing/android/src/main/java/org/iota/mobile/Converter.java]: renamed in https://github.com/iotaledger/iota.js/pull/355/commits/a09e7908381b1ed8dab7d0b7f7c1327f23455f89

**Pull request merged**